### ISSUE TITLE
GBC Pokemon Trade Fast Forward Lock Issues

### ISSUE THREAD

**Issue by JoopyJones** [mgba-emu/mgba#2720]
Hello there, I'd like to submit a repeatable bug that I just experienced while trading pokemon using the "New Multiplayer Window" option. I haven't been able to find another issue about this on git, so I figured I'd submit one.

Games: Pokemon Gen 2: Pokemon Gold, Pokemon Silver. Vanilla save files.
Action: Trading using the New Multiplayer Window option(Between 2 distinct saves).
Input: Load save1 (go to trading spot), click 'New Multiplayer Window', then load save2. then both trade in game.
Issue: Player 1 trading player2 will cause player2 to be stuck in fastdorward. This causes trades to fail since p1's search request will timeout before p2's trade request can be found.

Behavior: With two files that have not traded before - p1 can connect with, trade with, and disconnect with p2 as expected. Trades happen normally. Then, if p1 tries to trade with the same p2, p2 will be stuck in fastforward at the time the "would you like to trade? before opening the link you must...." prompt shows. If Both saves have not traded, it works as expected. if X traded with Y before, Y will fastforward. If Y trades with X it will work.. But then after Y trades with X, X will fast forward.... then any save that tires to trade with X or Y will fast forward.

x(v)->y(v) = success.
y(nV)-> x(nV) = success.
x(nV)->z(v) = fastforward.
z(v)->x(nV) or y(nV) = fastforward.

I have no idea why saves that have traded causes next trades to get stuck in fast forward..

Thanks

Joop

**Comment by endrift** [comment]
Please try turning on Sync to Video while trading. It's buggy with Sync to Audio.

Also I'll need the bug report zip if that doesn't work.

**Comment by JoopyJones** [comment]
Thanks for the quick reply! 

I updated my settings so that only Sync to Video was checked, but the same behavior occurred. I did find an odd solution that allowed for the trade to work - if you start capturing a gif (from p2's mgba client) before establishing the trade connection, the fast forward doesn't happen and you're able to trade; however, once the trade is completed p2 will get stuck in fast forward(regardless if the gif is still being captured). I would have attached the gif as well, but since it's writing out all of the fastforwarded frames, it takes forever just to view a single frame change.

I've attached a bug report zip with both of the roms and save files as well.

Hope this information helps, I appreciate you :) 

[trade_autoFastForward 2.zip](https://github.com/mgba-emu/mgba/files/9973580/trade_autoFastForward.2.zip)

**Comment by Ninbura** [comment]
I'm having the same issue with Pokemon Yellow, as soon as you initiate a trade with the multiplayer/secondary window the game gets stuck in fast forward. Tried on multiple computers with different architectures (TRX40, Z690) and the problem persists, I am running Windows 11 on both machines however.

Here's a video example:
https://user-images.githubusercontent.com/58058942/204167059-64661475-965f-4b9f-9f88-779007a5802e.mp4

As can be seen, the audio sync setting is checked, syncing to video or both audio & video does not remedy the issue.

**Comment by endrift** [comment]
Ohhh, I think I know what's going on here. Does it do the same thing if you initiate the trade from the first player?

**Comment by ninbura** [comment]
No, when the trade is initiated as first player it runs as expected, no fast forward.

**Comment by endrift** [comment]
Yes, I think I know what's going on now. I'll take a look soon.
